### ISSUE TITLE
VB to C# Property related issue

### ISSUE THREAD

**Issue by Savjiyani** [icsharpcode/CodeConverter#211]
"VB -> C#: "

### Input code
```VB
Public ReadOnly Property ItemByFunction(ByVal func As CEnumDefine.EFunction) As CLoadConfig
            Get
                Dim oLoadConfig As CLoadConfig = Nothing
                For i As Integer = 0 To Me.Count - 1
                    If Me(i).FunctionKind = func Then
                        oLoadConfig = Me(i)
                        Exit For
                    End If
                Next

                Return oLoadConfig
            End Get
        End Property
```

### Erroneous output
```csharp
public CLoadConfig ItemByFunction
        {
            get
            {
                CLoadConfig oLoadConfig = null;
                var loopTo = this.Count - 1;
                for (int i = 0; i <= loopTo; i++)
                {
                    if (this[i].FunctionKind == func)
                    {
                        oLoadConfig = this[i];
                        break;
                    }
                }

                return oLoadConfig;
            }
        }
```

### Expected output
```csharp
public CLoadConfig ItemByFunction(CEnumDefine.EFunction func)
        {
            CLoadConfig oLoadConfig = null;
            for (int i = 0; i <= this.Count - 1; i++)
            {
                if (this[i].FunctionKind == func)
                {
                    oLoadConfig = this[i];
                    break;
                }
            }
            return oLoadConfig;
        }
```

### Details
VS extension 

Version in use: 6.1.0.
 
In older version 6.0.0

The output is quite relevant

And another problem is

In VB properties can call with parameter but in C# It's not possible
I have seen that you are using indexer, It is feasible for one property 
What if one class have more than One parameterized property in VB Code

Please help me with this

**Comment by vfle** [comment]
Duplicate of #200 .

**Comment by Savjiyani** [comment]
Thank you for your reply

But my suggestion : 
No It's not duplicate of #200 
I said it was near to actual conversion in older version may be in 6.0.0 - 21/08/2018 or in 5.9.0 - 01/08/2018
Please consider this place

**Comment by GrahamTheCoder** [comment]
Thanks @Savjiyani. I think the reason that the old behaviour seemed close was actually a bug (the code didn't understand properties) - never the intention. Therefore I think the problem we need to solve is the same in this and #200. Let's hold any further discussion on #200 since it's the earlier of the two. That way we don't end up with two people implementing the fix in parallel.
